### ISSUE TITLE
[<ImportedReactComponent("Foo", from="bar")>]

### ISSUE THREAD

**Issue by alfonsogarciacaro** [Zaid-Ajaj/Feliz#294]
We should add an additional attribute to model imported components and so the same transformations in the call site. So we can use either optional arguments or (anonymous) record props easily:

```fsharp
type ExternalModule =
    [<ImportedReactComponent("AnotherComponent", from="external-module")>]
    static member Component(name: string, flag: bool): ReactElement = jsNative

[<ImportedReactComponent("AnotherComponent", from="external-module")>]
let AnotherComponent (props: {| foo: int; bar: string |}): ReactElement = jsNative 
```

This is so I don't forget, I will try to send a PR myself.

**Comment by Zaid-Ajaj** [comment]
Oh that would actually be really cool 😍 

Why not use the same the attribute, though? 

```fs
// import { AnotherComponent } from "external-module"; 
[<ReactComponent(import="AnotherComponent", from="external-module")>]
let Component(name: string, flag: bool): ReactElement = jsNative

// import Component from "external-module"
[<ReactComponent(importDefaultFrom="external-module")>]
let Component(name: string, flag: bool): ReactElement = jsNative
``` 
I personally like this better, what do you think?

**Comment by alfonsogarciacaro** [comment]
That should work too 😄

**Comment by alfonsogarciacaro** [comment]
My future me just tells me there may be an issue in the future if we want to make plugins work in the REPL, because currently I'm using `System.Activator` to initialize the plugin and in Fable this only works with the primary constructor. But optional arguments are not allowed for attribute constructors so we need to use multiple constructors.

Anyways, it won't be possible to always avoid attributes with multiple constructors (we already use them) and I'm sure we can find a workaround when the time comes 😸

**Comment by VitalyBrusentsev** [comment]
In case you guys go with a separate attribute, I would suggest using a verb instead: `[<ImportReactComponent>]`.
This would keep it consistent with JS interop attributes (like `[<ImportDefault>]`).

**Comment by Zaid-Ajaj** [comment]
@VitalyBrusentsev I think I will give just  try new parameters inside the same `[<ReactComponent>]` a try instead of a new attribute

@alfonsogarciacaro I am trying it out right now

**Comment by Zaid-Ajaj** [comment]
@alfonsogarciacaro Published Feliz v1.25 which should allow you to write
```fs
[<ReactComponent(import="Hello", from="external-module")>]
let HelloWrapper (name: string) (children: ReactElement []) = Html.none
```
Where the body of the `HelloWrapper` returns `null` because we can't get rid of the declaration and on call-site, the imported function is used instead with the applied parameters
